src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,10 +36,40 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * Optional classpath-relative path to a yml file containing extension settings.
+     */
+    protected static final String EXTENSION_SETTINGS_PATH = "/sample/helloworld-settings.yml";
+
+    /**
+     * The extension settings include a name, host address, and port.
+     */
+    protected ExtensionSettings settings;
+
+    /**
+     * Instantiate this extension, initializing the connection settings and REST actions.
      */
     protected BaseExtension() {
```

Comment:
The constructor here should take an argument that lets this class find the `ExtensionSettings`. This is how the subclass will pass its settings in.  You will need one or two constructors:
1. Definitely include one that allows the user to just instantiate extension settings without file reading, e.g., the constructor in the subclass would call `super(new ExtensionSettings("hello-world", "127.0.0.1", "4532"));`
2. Allow the YAML path one of these two ways:
   - Create a `String` constructor and just pass in the path and do the file reading here
   - Pass the resolved path to the constructor you created in bullet 1, like `super(Extension.readSettingsFromYaml(EXTENSION_SETTINGS_PATH));`.  This keeps it to only one constructor, but also introduces complexity for the user.  So I think I'd prefer just passing the String.